pkg/kustomize/apply.go

```diff
@@ -56,7 +58,7 @@ func (s *Kustomizer) ApplyKustomizationPath(path string) {
 	if _, err := os.Stat(kustomization); !errors.Is(err, os.ErrNotExist) {
 		klog.Infof("Applying kustomization at %v ", kustomization)
 		if err := ApplyKustomizationWithRetries(path, s.kubeconfig); err != nil {
-			klog.Fatalf("Applying kustomization at %v failed: %s. Giving up.", kustomization, err)
+			klog.Errorf("Applying kustomization at %v failed: %s. Giving up.", kustomization, err)
```

Comment:
This is the only change needed. The call to `Fatalf` causes microshift to exit, which means the API is unavailable. Systemd eventually restarts microshift, and the cycle resumes. If this call is changed to `Errorf` then the error is logged, the kustomizer eventually times out and MicroShift goes into a ready state. We don't need the changes to the retry loop to fix the bug.